### PR TITLE
Send kUAModel and kUAPlatformVersion CHs when requested.

### PR DESCRIPTION
**Pull request by mkarolin**
Resolves https://github.com/brave/brave-browser/issues/29875

## Submitter Checklist:

- [ ] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Review by pes10k — APPROVED** [review]
This looks terrific! Did not realize such a (seemingly) small change would need so much code.  Thank you for sorting this out!

Just to verify, the initial build value for the clamped version will be “0”, but we could change it to something else (like “x”) using griffin?

**Comment by mkarolin** [comment]
Thanks @pes10k, a lot of the changes are for tests. Yes, the default value is set to "0", but the parameter type is a string, so it can take "x" (I used "x" in one of the tests).

File: chromium_src/components/embedder_support/user_agent_utils.cc [hunk 1]
```diff
@@ -18,5 +24,31 @@ constexpr char kBraveBrandNameForCHUA[] = "Brave";
...
+blink::UserAgentMetadata GetUserAgentMetadata(const PrefService* pref_service) {
+  blink::UserAgentMetadata metadata =
+      GetUserAgentMetadata_ChromiumImpl(pref_service);
+  if (base::FeatureList::IsEnabled(
+          blink::features::kClampPlatformVersionClientHint)) {
+    // Clamp platform version
+    base::Version platform_version(metadata.platform_version);
+    metadata.platform_version = base::StringPrintf(
+        "%d.%d.%s", platform_version.components()[0],
+        platform_version.components()[1],
```
Comment (goodov) [inline chromium_src/components/embedder_support/user_agent_utils.cc:47]: either add `CHECK(components().size() >= 2u)` or add a clamper function:
```
base::Version ClampVersionComponents(const base::Version& version) {
  std::vector<uint32_t> components(version.components());
  if (components.size() >= 3u) {
    components[2] = clamped;
    components.resize(3u);
  }
  return base::Version(std::move(components));
}
```

Comment (mkarolin) [reply chromium_src/components/embedder_support/user_agent_utils.cc:47]: Great point. Will fix.

**Review by goodov — APPROVED** [review]

**Comment by kjozwiak** [comment]
Verification **PASSED** on `Win 11 x64` using the following build(s):

```
Brave | 1.53.4 Chromium: 113.0.5672.53 (Official Build) nightly (64-bit)
-- | --
Revision | 12f5dac35d12e8f4e72d7dd11df557ef93bc046f-refs/branch-heads/5672@{#703}
OS | Windows 11 Version 22H2 (Build 22621.1555)
```

Using the STR/Cases outlined via https://github.com/brave/brave-browser/issues/29875#issue-1675788007, reproduced the using `1.52.77 Chromium: 113.0.5672.53`. However, I could only reproduce the issue with [this link ](https://strava-web.app.link/3p?$3p=e_it&$original_url=https%3A%2F%2Fsupport%2Estrava%2Ecom%2Fhc%2Farticles%2F13013167258381%2DSubscription%2DPricing%2DFAQ%3F%24web%5Fonly%3Dtrue). Seems like [this link](https://rsci.app.link/syPp8BMP4yb?_p=c71729c39d0a7af2e11590f8ec) loaded fine.

**`1.52.77 Chromium: 113.0.5672.53`** | **`1.52.77 Chromium: 113.0.5672.53`**
-------------------------------------------|--------------------------------------------
reproduced0 | reproduced1

Using the STR/Cases mentioned above, verified that both websites are correctly loading as per the following:

**`Example`** | **`Example`** | **`Example`**
----------------|----------------|----------------
fixed1 | fixed2 | fixed3

**Pull request merged**